### DISCUSSION TITLE
Python Flask how to debug my code

### DISCUSSION THREAD

**Discussion by DJManas** [syl20bnr/spacemacs#15387] [Q&A]
Hello,

I had been trying to move to any kind of emacs, but I get problem with how to use it to debug Flask App in python. I had been able to run standard python script with DAP mode, but I am unable to find out how to launch Flask App and maybe restart debugging proces on changes.

This question is maybee a little general, but maybe someone can point me out to good tutorial. My GoogleFu seems short even going through documentation to DAP mode (I can't even read, what variables I can use in dap edit template command.

Thanks,
Regards,
Petr Sourek

**Comment by tshu-w** [comment]
Off topic comment, maybe you don't need to use dap or single step debugging, print and logging can solve most problems.

**Reply by DJManas** [reply]
To be honest. I have long time experience with print and logging, when I am developing Adobe PDF forms in SAP for one customer, where the customer don't have the license for Adobe Acrobat, which is needed for JS debugging and to be honest the programming gets a lot longer to pinpoint location of error, than to launch up live debugger and see the values in there. I am not telling its bad, but for me it's less efficient.

**Comment by lebensterben** [comment]
You can join their discord channel and ask this question there.

See https://github.com/emacs-lsp/dap-mode

**Reply by DJManas** [reply]
Ok, I will try it, thanks.
